apps/api-extractor/src/collector/Collector.ts

```diff
@@ -306,6 +307,43 @@ export class Collector {
     return this._entitiesByAstEntity.get(astEntity);
   }
 
+  /**
+   * Returns the name to emit for the given `symbol`.
+   * This is usually the `name` of that `symbol`.
+   * Only if this `symbol` has a `CollectorEntity`, the `nameForEmit` of that entity is used.
+   * @param symbol the symbol to compute the name to emit for
+   * @returns the name to emit for the given `symbol`
+   */
+  public getEmitName(symbol: ts.Symbol): string {
+    const collectorEntity: CollectorEntity | undefined = this._entitiesBySymbol.get(symbol);
+    return collectorEntity?.nameForEmit ?? symbol.name;
+  }
+
+  public getParentSymbols(symbol: ts.Symbol): ts.Symbol[] {
+    const result: ts.Symbol[] = [];
+    let currentSymbol: ts.Symbol | undefined = symbol;
+    while (currentSymbol) {
+      currentSymbol = this._getParentSymbol(currentSymbol);
+      if (currentSymbol) {
+        result.unshift(currentSymbol);
+      }
+    }
+    return result;
+  }
+
+  private _getParentSymbol(symbol: ts.Symbol): ts.Symbol | undefined {
+    const collectorEntity: CollectorEntity | undefined = this._entitiesBySymbol.get(symbol);
+    if (collectorEntity) {
+      const firstNamespaceImport: AstNamespaceImport | undefined = [
+        ...collectorEntity.astNamespaceImports
+      ][0];
```

Comment:
Thanks for the advance notice!

Reply:
To be ready to update this PR when your PRs are merged, I did the following:
* created an [integration branch for your two PRs](https://github.com/fwienber/rushstack/commits/zelliott/integrate)
* rebased and minimized my changes on top of that branch as a [new branch](https://github.com/fwienber/rushstack/commits/fwienber/api-extractor-fix-3593-reference-alias-rebased)
* created a [draft PR](https://github.com/fwienber/rushstack/pull/1) inside my fork so that you can review the diff to your PRs

Once your PRs are merged, it should be easy to rebase those changes onto main.

Reply:
I now also added commits that lead to a similar state to this PR.